nano/secure/ledger.hpp

```diff
@@ -15,6 +15,27 @@ class shared_ptr_block_hash final
 	size_t operator() (std::shared_ptr<nano::block> const &) const;
 	bool operator() (std::shared_ptr<nano::block> const &, std::shared_ptr<nano::block> const &) const;
 };
+class account_state final
+{
+public:
+	account_state () = default;
+	account_state (nano::account_info const &, std::shared_ptr<nano::block>, nano::block_sideband const &);
+	account_state (nano::transaction const &, nano::block_store &, nano::account const &);
+	account_state (nano::transaction const &, nano::block_store &, nano::account_info const &);
+	nano::uint256_union head () const;
+	nano::uint256_union rep () const;
+	nano::uint256_union open () const;
+	nano::amount balance () const;
```

Comment:
`head`, `rep`, `open` can all be `const &`, also could use the appropriate aliases we have currently, i.e `nano::block_hash`, `nano::account`; I am in the process of updating them all anyway though.